src/core/secondaryAdapters/officialOnyxiaApiClient.ts

```diff
@@ -305,34 +308,69 @@ export function createOfficialOnyxiaApiClient(params: {
-                    .then(({ data }) => data.catalogs),
+                    }>("/public/catalogs")
+                    .then(({ data }) => {
+                        for (let catalog of data.catalogs) {
+                            let entries: Entry[] = [];
+                            for (let key of Object.keys(catalog.catalog.entries)) {
+                                if (key !== "library-chart") {
+                                    let entry = {
+                                        name: key,
+                                        values: catalog.catalog.entries[
+                                            key as keyof typeof catalog.catalog.entries
+                                        ],
+                                    } as unknown as Entry;
+                                    entry.latest = () =>
+                                        [...entry.values]
+                                            .sort((a, b) =>
+                                                a.version.localeCompare(
+                                                    b.version,
+                                                    undefined,
+                                                    {
+                                                        numeric: true,
+                                                        sensitivity: "base",
+                                                    },
+                                                ),
+                                            )
+                                            .pop()!;
+                                    entries.push(entry);
+                                }
+                            }
+                            catalog.catalog.entries = entries;
+                        }
+                        return data.catalogs as Catalog[];
+                    }),
             { "promise": true },
         ),
         "getPackageConfig": ({ catalogId, packageName }) =>
             axiosInstance
-                .get<{
-                    config: JSONSchemaObject;
-                    sources?: string[];
-                    dependencies?: {
-                        name: string;
-                    }[];
-                }>(`/public/catalog/${catalogId}/${packageName}`)
+                .get<
+                    {
+                        config: JSONSchemaObject;
+                        sources?: string[];
+                        dependencies?: {
+                            name: string;
+                        }[];
+                    }[]
+                >(`/public/catalogs/${catalogId}/charts/${packageName}`)
                 .then(({ data }) => ({
-                    "dependencies": data.dependencies?.map(({ name }) => name) ?? [],
-                    "sources": data.sources ?? [],
+                    "dependencies": data[0].dependencies?.map(({ name }) => name) ?? [],
+                    "sources": data[0].sources ?? [],
```

Comment:
I see you have adapted this in a future PR.  
I assume the first element is always the newer chart.

Reply:
It is temporary. But yes, the API is supposed to return the data sorted but it is not mandatory